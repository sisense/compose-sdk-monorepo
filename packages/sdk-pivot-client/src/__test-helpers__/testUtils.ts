import EventEmitter from 'events';
import { TreeNode } from '../tree-structure/index.js';
import { treeNode } from '../tree-structure/utils/index.js';
import { Jaql, JaqlPanel, JaqlRequest } from '../data-load/types.js';
import { v4 as uuid } from 'uuid';

/**
 * @param {Function} emitter - instance of EventEmitter (node-events)
 * @param {string} methodNameCallEvent - name of custom event
 * @returns {void}
 */
export const emitterCallHandle = (
  emitter: EventEmitter,
  methodNameCallEvent: string,
): Promise<string> =>
  new Promise((res) => {
    emitter.once(methodNameCallEvent, () => {
      res(methodNameCallEvent);
    });
  });
/**
 *
 * @returns {Array} - returns callback listener and dispatcher (node events library)
 */
export const initEventEmitter = (): {
  awaitForCall: (methodName: string) => Promise<string>;
  emitCall: (methodName: string) => void;
} => {
  const emitter = new EventEmitter();
  const awaitForCall = (methodName: string) => emitterCallHandle(emitter, methodName);
  const emitCall = (methodName: string) => {
    emitter.emit(methodName);
  };
  return { awaitForCall, emitCall };
};

export const getNodeProp = (item: any, propName: string, list: Array<any> = []): Array<any> => {
  if (!item) {
    return list;
  }
  let children: Array<TreeNode> = [];

  if (Array.isArray(item)) {
    children = item;
  } else {
    children = treeNode.getChildren(item);
    list.push(item[propName]);
  }

  if (children && children.length) {
    children.forEach((child: Record<string, any>) => {
      getNodeProp(child, propName, list);
    });
  }
  return list;
};

export const getNodeNames = (item: any, list: string[] = []): string[] =>
  getNodeProp(item, 'value', list);

export const getNodeIndexes = (item: any, list: string[] = []): string[] =>
  getNodeProp(item, 'index', list);

export const createTypedPanel = (
  type: string,
  jaqlIndex = 0,
  additionalJaqlProps: Record<any, any> = {},
  additionalPanelProps: Record<any, any> = {},
) => {
  const title = `${type}-${jaqlIndex}`;
  const jaql: Jaql = {
    title,
    datatype: 'datatype',
    dim: `[${title}]`,
    ...additionalJaqlProps,
  };
  return {
    panel: type,
    field: { index: jaqlIndex, id: title },
    jaql,
    ...additionalPanelProps,
  };
};

export const createTypedPanels = (
  type: string,
  count: number,
  startIndex = 0,
  additionalJaqlProps: Record<string, any> = {},
) =>
  Array.from(Array(count - startIndex)).map((item, index): JaqlPanel => {
    const jaqlIndex = index + startIndex;
    const title = `${type}-${jaqlIndex}`;
    const jaql: Jaql = {
      title,
      datatype: 'datatype',
      dim: `[${title}]`,
      ...additionalJaqlProps,
    };
    return {
      panel: type,
      field: { index: jaqlIndex, id: title },
      jaql,
    };
  });

export const createTestJaql = (
  rows?: Array<JaqlPanel>,
  columns?: Array<JaqlPanel>,
  measures?: Array<JaqlPanel>,
  scope?: Array<JaqlPanel>,
  additionalJaqlProps: Record<string, any> = {},
): JaqlRequest => {
  let metadata: Array<JaqlPanel> = [];
  if (rows) {
    metadata = metadata.concat(rows);
  }
  if (columns) {
    metadata = metadata.concat(columns);
  }
  if (measures) {
    metadata = metadata.concat(measures);
  }
  if (scope) {
    metadata = metadata.concat(scope);
  }

  return {
    queryGuid: uuid(),
    metadata,
    grandTotals: {
      title: 'Grand total',
    },
    ...additionalJaqlProps,
  };
};

export const delay = (time = 0, resolveValue?: any, rejectValue?: any): Promise<any> =>
  new Promise((resolve, reject) => {
    setTimeout(() => {
      if (typeof rejectValue !== 'undefined') {
        reject(rejectValue);
      } else {
        resolve(resolveValue);
      }
    }, time);
  });
