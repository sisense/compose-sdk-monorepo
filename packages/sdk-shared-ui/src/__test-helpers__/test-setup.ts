import { vi } from 'vitest';

// Mock ResizeObserver for tests since it's not available in jsdom
global.ResizeObserver = vi.fn().mockImplementation(() => ({
  observe: vi.fn(),
  unobserve: vi.fn(),
  disconnect: vi.fn(),
}));
