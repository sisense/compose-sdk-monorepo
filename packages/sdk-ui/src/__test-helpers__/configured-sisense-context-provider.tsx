import { mockToken, mockUrl } from '@/__mocks__/msw';
import { SisenseContextProviderProps } from '@/props';
import { SisenseContextProvider } from '@/sisense-context/sisense-context-provider';

const contextProviderProps: SisenseContextProviderProps = {
  url: mockUrl,
  token: mockToken,
  defaultDataSource: 'Sample ECommerce',
  appConfig: {
    queryCacheConfig: { enabled: false },
    trackingConfig: { enabled: false },
  },
};

export const ConfiguredSisenseContextProvider: React.FC<{
  children: React.ReactNode;
}> = ({ children }) => {
  return <SisenseContextProvider {...contextProviderProps}>{children}</SisenseContextProvider>;
};
