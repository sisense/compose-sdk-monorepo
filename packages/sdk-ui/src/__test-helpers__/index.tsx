/**
 * Common test utilities
 */
import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderResult } from '@testing-library/react';
import type { Cell, Data } from '@ethings-os/sdk-data';
import isObject from 'lodash-es/isObject';
import userEvent from '@testing-library/user-event';
import { SisenseContext, SisenseContextPayload } from '../sisense-context/sisense-context';
import { Authenticator, HttpClient } from '@ethings-os/sdk-rest-client';
import { ClientApplication } from '../app/client-application';
import { DeepPartial } from 'ts-essentials';
import { useTranslation } from 'react-i18next';
import { translation } from '../translation/resources/en';
import { CustomWidgetsProvider } from '../custom-widgets-provider';
import { MenuProvider } from '@/common/components/menu/menu-provider';

type UserSetupFn = (typeof userEvent)['setup'];
interface SetupResult extends RenderResult {
  user: ReturnType<UserSetupFn>;
}
/**
 * This helper method combines the userEvent.setup() and render() calls at the
 * start of a unit test using @testing-library/react.
 *
 * Based on recommendation by: https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent
 *
 * @param jsx - React component instance to test, passed directly to render()
 * @param useFakeTimers - true if Vitest fake timers are enabled
 * @returns
 */
export function setup(jsx: ReactElement, useFakeTimers = false): SetupResult {
  return {
    // Need to set "advanceTimers" if we're using Vitest's fake timers.
    // The solution in https://github.com/testing-library/user-event/issues/833#issuecomment-1171452841
    // is for Jest, but this seems to work for Vitest as well.
    user: userEvent.setup(
      useFakeTimers ? { advanceTimers: (ms) => vi.advanceTimersByTime(ms) } : undefined,
    ),
    ...render(jsx),
  };
}

export const MockedHighchartsReact = ({
  options,
}: {
  /**
   * Highcharts chart configuration object.
   * Please refer to the Highcharts (API documentation)[https://api.highcharts.com/highcharts/].
   */
  options: object;
}) => {
  return <div>{JSON.stringify(options)}</div>;
};

type MockedSisenseContextProviderProps = PropsWithChildren<
  DeepPartial<Omit<SisenseContextPayload, 'app'>>
>;

const mockHttpClient = new HttpClient(
  'http://mock-url/sometenant?someparam=true',
  {} as Authenticator,
  'mock-env',
);
export const MockedSisenseContextProvider = ({
  children,
  tracking = {
    enabled: true,
  },
  isInitialized = true,
  errorBoundary,
}: MockedSisenseContextProviderProps) => {
  return (
    <SisenseContext.Provider
      value={{
        app: {
          httpClient: mockHttpClient,
        } as ClientApplication,
        tracking: {
          enabled: tracking.enabled ?? true,
          packageName: tracking.packageName ?? 'sdk-ui',
        },
        isInitialized,
        errorBoundary: { showErrorBox: true, ...errorBoundary },
      }}
    >
      <MenuProvider>
        <CustomWidgetsProvider>{children}</CustomWidgetsProvider>
      </MenuProvider>
    </SisenseContext.Provider>
  );
};

/**
 * Extends the provided data by blurring specified rows.
 *
 * @param {Data} data - The original data object.
 * @param {number[]} blurRowsIndexes - An array of row indexes to be blurred.
 * @returns {Data} The data object with blurred rows.
 */
export const withBlurredRows = (data: Data, blurRowsIndexes: number[]) => ({
  ...data,
  rows: data.rows.map((row, rowIndex) =>
    row.map((value) => {
      const cell: Cell = isObject(value) ? value : { data: value };

      if (blurRowsIndexes.includes(rowIndex)) {
        return { ...cell, blur: true };
      }
      return cell;
    }),
  ),
});

/**
 * Mocking the i18next translation functionality for tests.
 */
export const setupI18nMock = () => {
  vi.mock('react-i18next', async () => {
    const original: { useTranslation: typeof useTranslation } = await vi.importActual(
      'react-i18next',
    );

    // Helper function to get nested translations safely
    const getNestedTranslation = (key: string): string | undefined => {
      return key.split('.').reduce((accumulator: any, currentKey: string) => {
        if (accumulator && typeof accumulator === 'object' && currentKey in accumulator) {
          return accumulator[currentKey];
        }
        return undefined;
      }, translation) as string | undefined;
    };

    return {
      ...original,
      useTranslation: () => ({
        ...original.useTranslation(),
        t: vi.fn((key, options?: Record<string, any>) => {
          const translatedValue = getNestedTranslation(key);
          if (translatedValue && options) {
            // Replace variables in the translated string
            return translatedValue.replace(/{{(.*?)}}/g, (_, varName) => {
              const trimmedVarName = varName.trim();
              return options[trimmedVarName] || `{{${trimmedVarName}}}`; // Fallback if variable is not found
            });
          }
          return translatedValue ?? key; // Fallback to the key if translation is not found
        }),
      }),
    };
  });
};
