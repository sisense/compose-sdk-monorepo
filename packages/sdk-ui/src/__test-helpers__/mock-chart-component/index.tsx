// Mocks the Chart component and provides a way to access the rendered charts.
// Use this to simulate user interactions with the chart while actual Chart component is mocked.

import { ChartProps } from '@/props';
import { RenderedChartModel } from './rendered-chart-model';

export type ChartMocksManager = {
  /**
   * Replace the actual Chart component with a mocked version.
   */
  mockChartComponent: () => void;
  /**
   * Restore the actual Chart component.
   */
  unmockChartComponent: () => void;
  /**
   * List of models of rendered charts.
   * Use these models to simulate user interactions with the rendered charts.
   */
  renderedCharts: RenderedChartModel[];
  /**
   * Clear all rendered charts from the mocks.
   */
  clearMocks: () => void;
};

const renderedCharts: RenderedChartModel[] = [];

const mockingConfig = {
  shouldChartBeMocked: false,
};

vi.mock('@/chart', async () => {
  const actualModule = await vi.importActual<typeof import('@/chart')>('@/chart');

  return {
    ...actualModule,
    Chart: (props: ChartProps) => {
      if (!mockingConfig.shouldChartBeMocked) {
        return actualModule.Chart(props); // Use the actual Chart component
      }
      const chartModel = new RenderedChartModel(
        props,
        `Chart_${renderedCharts.length} (${props.chartType})`,
      );
      renderedCharts.push(chartModel);
      return chartModel.render();
    },
  };
});

const mockChartComponent = () => {
  clearMocks();
  mockingConfig.shouldChartBeMocked = true;
};

const unmockChartComponent = () => {
  mockingConfig.shouldChartBeMocked = false;
};

const clearMocks = () => {
  renderedCharts.splice(0, renderedCharts.length);
};

export const chartMocksManager: ChartMocksManager = {
  mockChartComponent,
  unmockChartComponent,
  clearMocks,
  renderedCharts,
};
