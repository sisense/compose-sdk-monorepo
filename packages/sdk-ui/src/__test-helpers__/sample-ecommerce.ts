import {
  Dimension,
  DateDimension,
  Attribute,
  createAttribute,
  createDateDimension,
  createDimension,
} from '@sisense/sdk-data';

export const DataSource = 'Sample ECommerce';

interface BrandDimension extends Dimension {
  Brand: Attribute;
  BrandID: Attribute;
}
export const Brand = createDimension({
  name: 'Brand',
  Brand: createAttribute({
    name: 'Brand',
    type: 'text-attribute',
    expression: '[Brand.Brand]',
  }),
  BrandID: createAttribute({
    name: 'BrandID',
    type: 'numeric-attribute',
    expression: '[Brand.Brand ID]',
  }),
}) as BrandDimension;

interface CategoryDimension extends Dimension {
  Category: Attribute;
  CategoryID: Attribute;
}
export const Category = createDimension({
  name: 'Category',
  Category: createAttribute({
    name: 'Category',
    type: 'text-attribute',
    expression: '[Category.Category]',
  }),
  CategoryID: createAttribute({
    name: 'CategoryID',
    type: 'numeric-attribute',
    expression: '[Category.Category ID]',
  }),
}) as CategoryDimension;

interface CommerceDimension extends Dimension {
  AgeRange: Attribute;
  BrandID: Attribute;
  CategoryID: Attribute;
  Condition: Attribute;
  Cost: Attribute;
  CountryID: Attribute;
  DateMonth: Attribute;
  Gender: Attribute;
  Quantity: Attribute;
  Revenue: Attribute;
  VisitID: Attribute;
  Date: DateDimension;
}
export const Commerce = createDimension({
  name: 'Commerce',
  AgeRange: createAttribute({
    name: 'AgeRange',
    type: 'text-attribute',
    expression: '[Commerce.Age Range]',
  }),
  BrandID: createAttribute({
    name: 'BrandID',
    type: 'numeric-attribute',
    expression: '[Commerce.Brand ID]',
  }),
  CategoryID: createAttribute({
    name: 'CategoryID',
    type: 'numeric-attribute',
    expression: '[Commerce.Category ID]',
  }),
  Condition: createAttribute({
    name: 'Condition',
    type: 'text-attribute',
    expression: '[Commerce.Condition]',
  }),
  Cost: createAttribute({
    name: 'Cost',
    type: 'numeric-attribute',
    expression: '[Commerce.Cost]',
  }),
  CountryID: createAttribute({
    name: 'CountryID',
    type: 'numeric-attribute',
    expression: '[Commerce.Country ID]',
  }),
  DateMonth: createAttribute({
    name: 'DateMonth',
    type: 'numeric-attribute',
    expression: '[Commerce.Date (Month)]',
  }),
  Gender: createAttribute({
    name: 'Gender',
    type: 'text-attribute',
    expression: '[Commerce.Gender]',
  }),
  Quantity: createAttribute({
    name: 'Quantity',
    type: 'numeric-attribute',
    expression: '[Commerce.Quantity]',
  }),
  Revenue: createAttribute({
    name: 'Revenue',
    type: 'numeric-attribute',
    expression: '[Commerce.Revenue]',
  }),
  VisitID: createAttribute({
    name: 'VisitID',
    type: 'numeric-attribute',
    expression: '[Commerce.Visit ID]',
  }),
  Date: createDateDimension({
    name: 'Date',
    expression: '[Commerce.Date (Calendar)]',
  }),
}) as CommerceDimension;

interface CountryDimension extends Dimension {
  Country: Attribute;
  CountryID: Attribute;
}
export const Country = createDimension({
  name: 'Country',
  Country: createAttribute({
    name: 'Country',
    type: 'text-attribute',
    expression: '[Country.Country]',
  }),
  CountryID: createAttribute({
    name: 'CountryID',
    type: 'numeric-attribute',
    expression: '[Country.Country ID]',
  }),
}) as CountryDimension;
