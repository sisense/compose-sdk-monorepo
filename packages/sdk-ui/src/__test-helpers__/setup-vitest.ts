import matchers, { TestingLibraryMatchers } from '@testing-library/jest-dom/matchers';
import { cleanup } from '@testing-library/react';
import 'vitest-canvas-mock';

import createFetchMock from 'vitest-fetch-mock';
import { vi } from 'vitest';

// Add types for matchers from @testing-library/jest-dom
// https://github.com/testing-library/jest-dom/issues/439#issuecomment-1536524120
declare module 'vitest' {
  interface Assertion<T = any> extends jest.Matchers<void, T>, TestingLibraryMatchers<T, void> {}
}

// Manually extend Vitest's expect() with methods from @testing-library/jest-dom
expect.extend(matchers);

// Clean up mounted components after each test
// https://testing-library.com/docs/react-testing-library/api/#cleanup
afterEach(() => {
  cleanup();
});

vi.stubGlobal('__PACKAGE_VERSION__', '0.0.0');

// Mock ResizeObserver
const ResizeObserverMock = vi.fn(() => ({
  observe: vi.fn(),
  unobserve: vi.fn(),
  disconnect: vi.fn(),
}));

vi.stubGlobal('ResizeObserver', ResizeObserverMock);

// enable fetch mocking but don't mock anything by default
// and let tests decide to mock or not
const fetchMocker = createFetchMock(vi);
fetchMocker.enableMocks();
fetchMocker.dontMock();
