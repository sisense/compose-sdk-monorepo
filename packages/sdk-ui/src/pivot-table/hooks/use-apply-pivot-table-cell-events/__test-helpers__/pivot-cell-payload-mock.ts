import { type PivotTableDataOptionsInternal } from '@/chart-data-options/types';

import {
  AbstractDataTreeNode,
  ColumnDataNode,
  PivotTableCellPayload,
  RowDataNode,
  ValueDataNode,
} from '../types';

export const createMockDataOptions = (): PivotTableDataOptionsInternal =>
  ({
    rows: [{ column: { name: 'Category' } }, { column: { name: 'SubCategory' } }],
    columns: [{ column: { name: 'Year' } }, { column: { name: 'Quarter' } }],
    values: [{ column: { name: 'Sales' } }, { column: { name: 'Profit' } }],
  } as PivotTableDataOptionsInternal);

export const createMockTreeNode = <T extends AbstractDataTreeNode = AbstractDataTreeNode>(
  value: string,
  content: string,
  level: number,
  metadataType: string,
  userType: string,
  index?: number,
  parent?: T,
): T =>
  ({
    value,
    content,
    level,
    metadataType,
    userType,
    parent,
    ...(index !== undefined && { index }),
  } as T);

export const createMockCellPayload = (
  eventType: 'click' | 'contextmenu' = 'click',
): PivotTableCellPayload => {
  const rowParent = createMockTreeNode<RowDataNode>(
    'Electronics',
    'Electronics',
    0,
    'rows',
    'data',
  );
  const rowTreeNode = createMockTreeNode<RowDataNode>(
    'Laptops',
    'Laptops',
    1,
    'rows',
    'data',
    1,
    rowParent,
  );

  const columnParent = createMockTreeNode<ColumnDataNode>('2023', '2023', 0, 'columns', 'data');
  const columnTreeNode = createMockTreeNode<ColumnDataNode>(
    'Q1',
    'Q1',
    1,
    'columns',
    'data',
    1,
    columnParent,
  );

  const measureTreeNode = createMockTreeNode<ValueDataNode>(
    'Sales',
    '$50,000',
    0,
    'measures',
    'measureBottom',
    0,
  );

  const dataNode = createMockTreeNode<ValueDataNode>('50000', '$50,000', 0, 'values', 'data', 0);

  return {
    dataNode,
    rowTreeNode,
    columnTreeNode,
    measureTreeNode,
    isDataCell: true,
    event: new MouseEvent(eventType),
  };
};
