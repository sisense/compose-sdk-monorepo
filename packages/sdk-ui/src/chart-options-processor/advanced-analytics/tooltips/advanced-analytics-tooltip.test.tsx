import { screen } from '@testing-library/dom';
import { render } from '@testing-library/react';

import { MockedSisenseContextProvider } from '@/__test-helpers__';

import ForecastToolipComponent from './forecast-tooltip';
import TrendToolipComponent from './trend-tooltip';

const forecastTooltipProps = {
  confidenceValue: '80%',
  lowerValue: '1.01M',
  forecastValue: '2.01M',
  upperValue: '3.01M',
  title: 'Total Cost',
  x1Value: '2.01k',
};

const trendTooltipProps = {
  x1Value: '2.01k',
  modelType: 'logarithmic',
  title: 'Total Cost',
  trendData: {
    min: '190.0K',
    max: '4.84M',
    median: '2.28M',
    average: '2.15M',
  },
  localValue: '2.01M',
};

const withContext = (children: React.ReactNode) => (
  <MockedSisenseContextProvider>{children}</MockedSisenseContextProvider>
);

describe('ForecastTooltipComponent', () => {
  it('should render the title and values correctly', () => {
    render(withContext(<ForecastToolipComponent {...forecastTooltipProps} />));

    expect(screen.getByText('Total Cost')).toBeInTheDocument();
    expect(screen.getByText('80%')).toBeInTheDocument();
    expect(screen.getByText('1.01M')).toBeInTheDocument();
    expect(screen.getByText('2.01M')).toBeInTheDocument();
    expect(screen.getByText('3.01M')).toBeInTheDocument();
    expect(screen.getByText('2.01k')).toBeInTheDocument();
  });

  it('should contain translation keys', () => {
    render(withContext(<ForecastToolipComponent {...forecastTooltipProps} />));
    expect(screen.getByText('Forecast')).toBeInTheDocument();
    expect(screen.getByText('Forecast Value')).toBeInTheDocument();
    expect(screen.getByText('Upper Bound')).toBeInTheDocument();
    expect(screen.getByText('Lower Bound')).toBeInTheDocument();
    expect(screen.getByText('Confidence Interval')).toBeInTheDocument();
  });
  it('should match the snapshot', () => {
    const { container } = render(
      withContext(<ForecastToolipComponent {...forecastTooltipProps} />),
    );
    expect(container).toMatchSnapshot();
  });
});

describe('TrendTooltipComponent', () => {
  it('should render the title, model type, and trend data correctly including translation keys', () => {
    render(withContext(<TrendToolipComponent {...trendTooltipProps} />));

    expect(screen.getByText('Trend')).toBeInTheDocument();
    expect(screen.getByText('Total Cost')).toBeInTheDocument();
    expect(screen.getByText('Type')).toBeInTheDocument();
    expect(screen.getByText('Logarithmic Trend')).toBeInTheDocument();
    expect(screen.getByText('Min 190.0K')).toBeInTheDocument();
    expect(screen.getByText('Max 4.84M')).toBeInTheDocument();
    expect(screen.getByText('Median 2.28M')).toBeInTheDocument();
    expect(screen.getByText('Average 2.15M')).toBeInTheDocument();
  });

  it('should render the local value and x1 value correctly', () => {
    render(withContext(<TrendToolipComponent {...trendTooltipProps} />));

    // Testing for x1 value and local value
    expect(screen.getByText('2.01M')).toBeInTheDocument();
    expect(screen.getByText('2.01k')).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { container } = render(withContext(<TrendToolipComponent {...trendTooltipProps} />));

    expect(container).toMatchSnapshot();
  });
});
