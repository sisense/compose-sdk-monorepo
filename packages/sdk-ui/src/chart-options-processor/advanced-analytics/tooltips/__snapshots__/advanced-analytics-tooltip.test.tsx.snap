// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ForecastTooltipComponent > should match the snapshot 1`] = `
<div>
  <div
    style="color: rgb(91, 99, 114); font-size: 13px; line-height: 18px; margin: 4px 6px;"
  >
    <div
      style="display: flex; margin-bottom: 5px;"
    >
      <div
        style="flex-shrink: 0;"
      >
        Forecast
      </div>
      <div
        style="margin-left: 5px; font-weight: 600; overflow: hidden; text-overflow: ellipsis; word-wrap: normal; word-break: keep-all; white-space: nowrap;"
      >
        Total Cost
      </div>
    </div>
    <div
      style="display: table;"
    >
      <div
        style="display: table-row-group;"
      >
        <div
          style="display: table-row; height: 18px;"
        >
          <div
            style="display: table-cell; overflow: hidden; text-overflow: ellipsis; word-wrap: normal; word-break: keep-all; white-space: nowrap;"
          >
            Forecast Value
          </div>
          <div
            style="display: table-cell; padding-left: 15px; font-weight: 600;"
          >
            <div>
              2.01M
            </div>
          </div>
        </div>
      </div>
      <div
        style="display: table-row-group;"
      >
        <div
          style="display: table-row; height: 18px;"
        >
          <div
            style="display: table-cell; overflow: hidden; text-overflow: ellipsis; word-wrap: normal; word-break: keep-all; white-space: nowrap;"
          >
            Upper Bound
          </div>
          <div
            style="display: table-cell; padding-left: 15px; font-weight: 600;"
          >
            <div>
              3.01M
            </div>
          </div>
        </div>
      </div>
      <div
        style="display: table-row-group;"
      >
        <div
          style="display: table-row; height: 18px;"
        >
          <div
            style="display: table-cell; overflow: hidden; text-overflow: ellipsis; word-wrap: normal; word-break: keep-all; white-space: nowrap;"
          >
            Lower Bound
          </div>
          <div
            style="display: table-cell; padding-left: 15px; font-weight: 600;"
          >
            <div>
              1.01M
            </div>
          </div>
        </div>
      </div>
      <div
        style="display: table-row-group;"
      >
        <div
          style="display: table-row; height: 18px;"
        >
          <div
            style="display: table-cell; overflow: hidden; text-overflow: ellipsis; word-wrap: normal; word-break: keep-all; white-space: nowrap;"
          >
            Confidence Interval
          </div>
          <div
            style="display: table-cell; padding-left: 15px; font-weight: 600;"
          >
            <div>
              80%
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="font-size: 10px; color: rgb(102, 102, 102); margin-top: 6px;"
    >
      <hr
        class="csdk-border-t"
        style="margin: 7px 0px;"
      />
    </div>
    2.01k
  </div>
</div>
`;

exports[`TrendTooltipComponent > should match the snapshot 1`] = `
<div>
  <div
    style="color: rgb(91, 99, 114); font-size: 13px; line-height: 18px; margin: 4px 6px;"
  >
    <div
      style="display: flex; margin-bottom: 5px;"
    >
      <div
        style="flex-shrink: 0;"
      >
        Trend
      </div>
      <div
        style="margin-left: 5px; font-weight: 600; overflow: hidden; text-overflow: ellipsis; word-wrap: normal; word-break: keep-all; white-space: nowrap;"
      >
        Total Cost
      </div>
    </div>
    <div
      style="display: table;"
    >
      <div
        style="display: table-row-group;"
      >
        <div
          style="display: table-row; height: 18px;"
        >
          <div
            style="display: table-cell; overflow: hidden; text-overflow: ellipsis; word-wrap: normal; word-break: keep-all; white-space: nowrap;"
          >
            Type
          </div>
          <div
            style="display: table-cell; padding-left: 15px; font-weight: 600;"
          >
            <div>
              Logarithmic Trend
            </div>
          </div>
        </div>
      </div>
      <div
        style="display: table-row-group;"
      >
        <div
          style="display: table-row; height: 18px;"
        >
          <div
            style="display: table-cell; overflow: hidden; text-overflow: ellipsis; word-wrap: normal; word-break: keep-all; white-space: nowrap;"
          >
            Local Value
          </div>
          <div
            style="display: table-cell; padding-left: 15px; font-weight: 600;"
          >
            <div>
              2.01M
            </div>
          </div>
        </div>
      </div>
      <div
        style="display: table-row-group;"
      >
        <div
          style="display: table-row; height: 18px;"
        >
          <div
            style="display: table-cell; overflow: hidden; text-overflow: ellipsis; word-wrap: normal; word-break: keep-all; white-space: nowrap;"
          >
            Trend Data
          </div>
          <div
            style="display: table-cell; padding-left: 15px; font-weight: 600;"
          >
            <div>
              Min 190.0K
            </div>
            <div>
              Max 4.84M
            </div>
            <div>
              Median 2.28M
            </div>
            <div>
              Average 2.15M
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="font-size: 10px; color: rgb(102, 102, 102); margin-top: 6px;"
    >
      <hr
        class="csdk-border-t"
        style="margin: 7px 0px;"
      />
    </div>
    2.01k
  </div>
</div>
`;
